### PR TITLE
Add check-external-dns-services

### PR DESCRIPTION
**Pull request by metrictwo**
## Pull Request Checklist
Example run:
```
/ # /etc/sensu/plugins/check-external-dns-services.rb --in-cluster --in-namespace
KubeExternalDNSServices CRITICAL: The following failures were detected: {"kube-system/bwilson"=>"bwilson2.k8s.sendgrid.net is missing IPs [172.16.23.3]"}
```

### CODE REVIEW

File: bin/check-kube-external-dns-services.rb [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+      next unless config[:services].empty? || config[:services].include?(s.metadata.name)
+
+      hostnames = hostnames_for_service(s)
+      next if hostnames.empty?
+
+      addresses = addresses_for_service(s)
+      next if addresses.empty?
+
+      msg_parts = []
+      hostnames.each do |hostname|
```
Comment (davidblum) [inline bin/check-kube-external-dns-services.rb]: Can we do `h` here for the iterator so its consistent with your `.each` services block above?

File: bin/check-kube-external-dns-services.rb [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        msg_parts << "#{display_hostname} has extra IPs #{extra_ips.inspect}" unless extra_ips.empty?
+        msg_parts << "#{display_hostname} is missing IPs #{missing_ips.inspect}" unless missing_ips.empty?
+      end
+
+      next if msg_parts.empty?
+      service_display_name = "#{s.metadata.namespace}/#{s.metadata.name}"
+      bad_service_msg_map.merge!(service_display_name => msg_parts.join(', '))
+    end
+
+    critical "The following failures were detected: #{bad_service_msg_map.inspect}" unless bad_service_msg_map.empty?
```
Comment (davidblum) [inline bin/check-kube-external-dns-services.rb]: This block is a little terse and hard to read.

Can we stick with the fairly conventional sensu patern of `if, elsif`? for readability and general consistency.

File: bin/check-kube-external-dns-services.rb [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+require 'resolv'
+require 'sensu-plugins-kubernetes/cli'
+require 'sensu-plugins-kubernetes/cli/namespaced'
+
+class KubeExternalDNSServices < Sensu::Plugins::Kubernetes::CLI
+  include Sensu::Plugins::Kubernetes::NamespacedCLI
+
+  option :services,
+         description: 'List of services to check',
+         long: '--services',
```
Comment (davidblum) [inline bin/check-kube-external-dns-services.rb]: no short `-s` option?

File: bin/check-kube-external-dns-services.rb (hunk 1 above)
Comment (metrictwo) [reply bin/check-kube-external-dns-services.rb]: Pushed.

File: bin/check-kube-external-dns-services.rb (hunk 2 above)
Comment (metrictwo) [reply bin/check-kube-external-dns-services.rb]: pushed

**Review by davidblum — APPROVED** [review]
🦈 
💇‍♂️ 
🍻

**Review by davidblum — APPROVED** [review]
🌮 
🍨 
🍧

**Pull request merged**